pygmt/tests/test_clib_loading.py

```diff
@@ -64,18 +89,85 @@ def test_load_libgmt_with_a_bad_library_path(monkeypatch):
     assert check_libgmt(load_libgmt()) is None
 
 
-def test_clib_names():
+def test_load_libgmt_with_broken_libraries(monkeypatch):
     """
-    Make sure we get the correct library name for different OS names.
+    Test load_libgmt still works when a broken library is found.
     """
-    for linux in ["linux", "linux2", "linux3"]:
-        assert clib_names(linux) == ["libgmt.so"]
-    assert clib_names("darwin") == ["libgmt.dylib"]
-    assert clib_names("win32") == ["gmt.dll", "gmt_w64.dll", "gmt_w32.dll"]
-    for freebsd in ["freebsd10", "freebsd11", "freebsd12"]:
-        assert clib_names(freebsd) == ["libgmt.so"]
-    with pytest.raises(GMTOSError):
-        clib_names("meh")
+    # load the GMT library before mocking the ctypes.CDLL function
+    loaded_libgmt = load_libgmt()
+
+    def mock_ctypes_cdll_return(libname):
+        """
+        Mock the return value of ctypes.CDLL.
+
+        Parameters
+        ----------
+        libname : str or FakedLibGMT or ctypes.CDLL
+            Path to the GMT library, a faked GMT library or a working library
+            loaded as ctypes.CDLL.
+
+        Return
+        ------
+        object
+            Either the loaded GMT library or the faked GMT library.
+        """
+        if isinstance(libname, FakedLibGMT):
+            # libname is a faked GMT library, return the faked library
+            return libname
+        if isinstance(libname, str):
```

Comment:
```suggestion
        elif isinstance(libname, str):
```

Reply:
pylint doesn't like this change.

Reply:
Oh ok, must be the return statement above, Ignore this then.